### PR TITLE
[NO-29][Training]Restore-code

### PR DESCRIPTION
**Pull request by loitq**
- [x] Reset commit merge pull request 21: f8051435a4240b760a72b9369680d74c0f31585f
- [x] Move all commit from branch feature/refactor-folder-view

### CODE REVIEW

**Comment by phantuan1803** [comment]
LGTM

**Review by phantuan1803 — APPROVED** [review]

File: app/Http/Controllers/Admin/AdminController.php [hunk 1]
```diff
@@ -15,21 +17,161 @@ class AdminController extends Controller
...
+        $user->email = $request->email;
+        $user->password = bcrypt($request->password);
+        if ($request->can_see === "on") {
+            $user->can_see = \App\User::IS_TRUE;
+        } 
+        if ($request->can_delete === "on") {
+            $user->can_delete = \App\User::IS_TRUE;
+        }
+
+        $user->save();
```
Comment (ghost) [inline app/Http/Controllers/Admin/AdminController.php:97]: Not sure if save () is successful
=> So need to check more error case (use IF || Try .. cath)
 - Add TH return message error

File: app/Http/Controllers/Admin/AdminController.php [hunk 2]
```diff
@@ -15,21 +17,161 @@ class AdminController extends Controller
...
+        } else {
+            $user->can_see = \App\User::IS_TRUE;
+        }
+        if ($request->can_delete != "on") {
+            $user->can_delete = \App\User::IS_FALSE;
+        } else {
+            $user->can_delete = \App\User::IS_TRUE;
+        }
+
+        $user->save();
```
Comment (ghost) [inline app/Http/Controllers/Admin/AdminController.php:145]: Not sure if save () is successful
=> So need to check more error case (use IF || Try .. cath)
 - Add TH return message error

File: app/Http/Controllers/Admin/AdminController.php [hunk 3]
```diff
@@ -15,21 +17,161 @@ class AdminController extends Controller
...
+     * Delete account user
+     *
+     * @param $id Id user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function handleDelete($id)
+    {
+        $user = User::find($id);
+        $user->delete();
```
Comment (ghost) [inline app/Http/Controllers/Admin/AdminController.php:160]: Not sure if delete() is successful
=> So need to check more error case (use IF || Try .. cath)
 - Add TH return message error

File: app/Http/Controllers/User/BlogController.php [hunk 4]
```diff
@@ -14,6 +14,36 @@ class BlogController extends Controller
...
+    {
+        $request->validate([
+            'title' => 'required',
+            'content' => 'required',
+        ]);
+        $blog = new Blog;
+        $blog->user_id = $this->defineUser()->id;
+        $blog->title = $request->title;
+        $blog->content = $request->content;
+        $blog->save();
```
Comment (ghost) [inline app/Http/Controllers/User/BlogController.php:46]: Not sure if save() is successful
=> So need to check more error case (use IF || Try .. cath)
 - Add TH return message error

File: app/Http/Controllers/Admin/AdminController.php (hunk 1 above)
Comment (loitq) [reply app/Http/Controllers/Admin/AdminController.php:97]: @phantuan1803 help me resovle in branch current

File: app/Http/Controllers/Admin/AdminController.php (hunk 2 above)
Comment (loitq) [reply app/Http/Controllers/Admin/AdminController.php:145]: @phantuan1803 ... help me

File: app/Http/Controllers/Admin/AdminController.php (hunk 3 above)
Comment (loitq) [reply app/Http/Controllers/Admin/AdminController.php:160]: @phantuan1803 can you resolve in new  pull request?

File: app/Http/Controllers/User/BlogController.php (hunk 4 above)
Comment (loitq) [reply app/Http/Controllers/User/BlogController.php:46]: @nguyentienthinh can you resolve in new pull request.

**Comment by ghost** [comment]
@phantuan1803 
@loitq 
Please fix comment

File: app/Http/Controllers/User/BlogController.php (hunk 4 above)
Comment (nguyentienthinh) [reply app/Http/Controllers/User/BlogController.php:46]: dạ cái này em đang fix ở branch của em á Anh

File: app/Http/Controllers/Admin/AdminController.php (hunk 1 above)
Comment (ghost) [reply app/Http/Controllers/Admin/AdminController.php:97]: wait for later updates

File: app/Http/Controllers/Admin/AdminController.php (hunk 2 above)
Comment (ghost) [reply app/Http/Controllers/Admin/AdminController.php:145]: wait for later updates

File: app/Http/Controllers/Admin/AdminController.php (hunk 3 above)
Comment (ghost) [reply app/Http/Controllers/Admin/AdminController.php:160]: wait for later updates

File: app/Http/Controllers/User/BlogController.php (hunk 4 above)
Comment (ghost) [reply app/Http/Controllers/User/BlogController.php:46]: wait for later updates

**Comment by ghost** [comment]
LGTM

**Review by hienl — APPROVED** [review]
LGTM

**Pull request merged**